index.js

```diff
@@ -30,129 +30,128 @@ function processFirstItem(stringList, callback) {
 
 /**
  * ### Challenge `processLength`
- * 
+ *
  * @instructions
  * PLEASE STUDY THE EXAMPLE CHALLENGE THOROUGHLY BEFORE PROCEEDING!
  * PLEASE STUDY THE EXAMPLE CHALLENGE THOROUGHLY BEFORE PROCEEDING!
- * 
+ *
  * Implement a higher-order function called `processLength`.
  * It takes two arguments:
  * @param list an array with elements of any type.
  * @param callback function that takes a number as its argument.
  * @returns the result of invoking `callback` passing the LENGTH of `list`.
- * 
+ *
  * Examples of usage of this higher-order function:
  * [1] Invoking `processLength` passing `['foo', 'bar']` and `(num) => num + 1000`,
  * should return 1002.
- * 
+ *
  * [2] Invoking `processLength` passing `[]` and `(num) => "There are " + num`,
  * should return "There are 0".
 */
-function processLength(/* CODE HERE */) {
-  /* CODE HERE */
+function processLength(list, callback) {
+  return callback(list.length);
 }
 
 /**
  * ### Challenge `processLastItem`
- * 
+ *
  * @instructions
  * Implement a higher-order function called `processLastItem`.
  * It takes two arguments:
  * @param stringList array of strings.
  * @param callback function that takes a string as its argument.
  * @returns the result of invoking `callback` with the LAST element in `stringList`.
- * 
+ *
  * Example of usage of this higher-order function:
  * Invoking `processLastItem` passing `['foo', 'bar']` and `(str) => str + str`,
  * should return 'barbar'.
 */
-function processLastItem(/* CODE HERE */) {
-  /* CODE HERE */
+function processLastItem(stringList, callback) {
+  return callback (stringList.pop());
 }
-
 /**
  * ### Challenge `processSum`
- * 
+ *
  * @instructions
  * Implement a higher-order function called `processSum`.
  * It takes two arguments:
  * @param numberList array of numbers.
  * @param callback function that takes a number as its argument.
  * @returns the result of invoking `callback` passing the SUM of all elements in `numberList`.
- * 
+ *
  * Examples of usage of this higher-order function:
  * [1] Invoking `processSum` passing `[10, 20, 30]` and `(num) => num + " is a big number!"`,
  * should return "60 is a big number!".
- * 
+ *
  * [2] Invoking `processSum` passing `[]` and `(num) => num + 1000`,
  * should return 1000.
 */
-function processSum(/* CODE HERE */) {
-  /* CODE HERE */
+function processSum(numberList, callback) {
+  return callback (numberList.reduce(function(acc, val) { return acc + val; }, 0)
+  )
```

Comment:
good job, try to use arrow functions when putting everything in one line like this.